4.1/crud-fields.md

```diff
@@ -2170,3 +2170,127 @@ Inside your custom field type, you can use these variables:
+CRUD::field('comments')->type('select_multiple');
+CRUD::field('comments')->type('select2_multiple');
+```
+
+- notes: 
+    - you _cannot_ use [the InlineCreate operation](/docs/{{version}}/crud-operation-inline-create) to have show a [+ Add Item] button next to the dropdown; that's because the Comment needs a `post_id` (not nullable); and until the Save button is clicked to submit the Create Post form, there is no `post_id`;
+    - you can use the `repeatable` field on the Post create/update form, to add multiple Comments in one go, but you'll also need to modify your `store()` and `update()` methods to take that into account, and transform the JSON received from the `repeatable` field into actual entries in the database (and back); check out [this example here](https://gist.github.com/tabacitu/4db255042a62c7458ee17fe33d65522c) for how that can be done for Invoice and InvoiceItem;
+    
+// TODO: create a tutorial on how to use the Repeatable field with exactly these models, Post and Comment;
+
+
+#### belongsToMany (n-n relationship)
+
+- example: 
+    - `User -> Role`
+    - a User has many Roles; a Role can also have many Users
+    - the foreign key is stored on a pivot table (usually the `user_roles` table has both `user_id` and `role_id`)
+- how to use:
+    - [the `belongsToMany` relationship should be properly defined](https://laravel.com/docs/eloquent-relationships#many-to-many) in both the User and Role models;
+    - you can add a dropdown on your User to pick the Roles that are connected to it; for that, use the [`relationship`](#relationship), [`select_multiple`](#select-multiple), [`select2_multiple`](#select2-multiple) or  [`select2_from_ajax_multiple`](#select2-from-ajax-multiple) fields;
+
+```php
+// inside UserCrudController::setupCreateOperation()
+CRUD::field('roles')->type('relationship');
+CRUD::field('roles')->type('select_multiple');
+CRUD::field('roles')->type('select2_multiple');
+
+// inside RoleCrudController::setupCreateOperation()
+CRUD::field('users')->type('relationship');
+CRUD::field('users')->type('select_multiple');
+CRUD::field('users')->type('select2_multiple');
+```
+
+- notes:
+    - if you choose the `relationship` field, you can also use [the InlineCreate operation](/docs/{{version}}/crud-operation-inline-create), which will add a [+ Add Item] button next to the dropdown, to let the admin create a Role in a modal, without leaving the current Create User form; 
+
+// TODO: what if you want to save extra information on the pivot table?
+
+#### hasOneThrough (1-1-1 relationship)
+
+// TODO
+// AFAIK it works perfectly fine, we should just point to `hasOne` above, right?
+
+#### hasManyThrough (1-1-n relationship)
+
+// TODO
+// AFAIK it works perfectly fine, we should just point to `hasMany` above, right?
+
+#### morphOne (1-1 polymorphic relationship)
+
+// TODO
+
+#### morphMany (1-n polymorphic relationship)
+
+// TODO
+
+#### morphToMany (n-n polymorphic relationship)
+
+// TODO
```

Comment:
We should make sure this new section includes ALL Laravel relationships, so that we leave no question unanswered. Whatever Eloquent relationship the dev wants to use, they'd find a corresponding field here.

@pxpm could you please take a look at these extra relationships, and let me know the fields that best make sense for each one? What you did in that email was WONDERFUL.

Reply:
This will probably address https://github.com/Laravel-Backpack/CRUD/issues/1926